pubsub/rabbitmq/rabbitmq.go

```diff
@@ -154,10 +154,10 @@ func (r *rabbitMQ) Init(_ context.Context, metadata pubsub.Metadata) error {
 
 	r.metadata = meta
 
-	r.reconnect(0)
-	// We do not return error on reconnect because it can cause problems if init() happens
```

Comment:
There’s a comment here that explains why it was done this way. Would changing it cause issues?

Reply:
I think the only issue is that the component would fail to initialize and the sidecar would crash, then the next time it would initialize correctly.

IMO that is desirable behavior.

I think the original code was actually hiding some potential bugs.